File: solutions_5756407898963968_0/C++/pgj92/magician.cpp
#include <iostream>
using namespace std;

int row[2][4];
int T;
int dummy;

void readInput(int no)
{
	int ans;

	cin >> ans;
	for (int i = 0; i < 4; i++)
		if (ans == i + 1)
			cin >> row[no][0] >> row[no][1] >> row[no][2] >> row[no][3];
		else
			cin >> dummy >> dummy >> dummy >> dummy;
}

void printr(int no)
{
	cout << row[no][0] << row[no][1] << row[no][2] << row[no][3] << endl;
}

void isIn(int r1[4], int r2[4])
{
	int card = 0;
	for (int i = 0; i < 4; i++)
		for (int j = 0; j < 4; j++)
			if (r1[i] == r2[j]) {
				if (card != 0) {
					cout << "Bad magician!" << endl;
					return;
				} else {
					card = r1[i];
				}
			}

	if (card == 0)
		cout << "Volunteer cheated!" << endl;
	else
		cout << card << endl;
}

void solve(int tc)
{
	readInput(0);
	readInput(1);

	cout << "Case #" << tc << ": ";
	isIn(row[0], row[1]);
}

int main()
{
	cin >> T;
	for (int i = 1; i <= T; i++) {
		solve(i);
	}

	return 0;
}

File: solutions_5756407898963968_0/Python/MadCow89/Magician.py
import math

def solve(input_path, output_path):
    with open(input_path, 'r') as fin:
        with open(output_path, 'w') as fout: 
            num_cases = int(fin.readline())
            for i in xrange(num_cases):
                appearance = [False for j in xrange(16)]
                result = -1
                row_idx = int(fin.readline())
                for j in xrange(4):
                    row = fin.readline()
                    if j == row_idx - 1:
                        for number in row.strip().split(" "):
                            appearance[int(number)-1] = True
                row_idx = int(fin.readline())
                for j in xrange(4):
                    row = fin.readline()
                    if j == row_idx - 1:
                        for number in row.strip().split(" "):
                            if appearance[int(number)-1]:
                                if result < 0:
                                    result = int(number)
                                else:
                                    result = 17
                            else:
                                appearance[int(number)-1] = True
                if 0 < result < 17:
                    answer = str(result)
                elif result < 0:
                    answer = "Volunteer cheated!"
                else:
                    answer = "Bad magician!"
                fout.write("Case #" + str(i+1) + ": " + answer + "\n")

dir = "F:\\Sync\\Dropbox\\Coding\\CodeJam\\2014\\"
input_path = dir + "A-small-attempt0.in"
output_path = dir + "A-small.out"
solve(input_path, output_path)
